### PR TITLE
ARTEMIS-856 Test fixes

### PR DESCRIPTION
**Pull request by clebertsuconic**

### CODE REVIEW

**Comment by clebertsuconic** [comment]
@michaelandrepearce I need you to take a look into this please.

MessageRedistributionFix is broken, up to the point I can't complete a whole testsuite on my internal CI.

My idea was the emulate the previous collection where redistributor would be an item, while still checking if redistributor is the only allowed item to be used on the loop.

I'm not sure how else we would change this without making substantial refactoring to your previous commit.


Can you look into this PR please?

**Comment by clebertsuconic** [comment]
@michaelandrepearce : But please don't merge it yet.. I'm running a whole testsuite on the CI before we can merge this... just let me know what you think first.

**Comment by clebertsuconic** [comment]
@michaelandrepearce Can you look into ExclusiveTest::testExclusiveTopicSharedConsumerFailover

It's failing even without my changes.. and after my changes.. I couldn't figure out today.

**Comment by michaelandrepearce** [comment]
@clebertsuconic, looks good, there we some bits with consumer lists with the redistributor i heavily want to avoid, im working an alternative option, but if to fix master happy to merge this, and i can supply my alternative later (have raised a pr so you can see, but still not complete)

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java [hunk 1]
```diff
@@ -544,6 +560,7 @@ public boolean isExclusive() {
 
    @Override
    public synchronized void setExclusive(boolean exclusive) {
+      new Exception("exclusive set at " + exclusive).printStackTrace();
```
Comment (michaelandrepearce) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java]: Think this needs to be removed.

Comment (clebertsuconic) [reply artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueueImpl.java]: Thanks.  That’s debug I forgot :p

**Comment by clebertsuconic** [comment]
This is replaced by #2211

**Pull request closed without merging**